source/game/kart/KartObject.cc

```diff
@@ -1,22 +1,52 @@
 #include "KartObject.hh"
 
+#include "game/kart/KartSub.hh"
+
 #include "game/system/RaceConfig.hh"
+#include "game/system/RaceManager.hh"
 
 namespace Kart {
 
-KartObject::KartObject(KartParam *param) : m_param(param) {}
+KartObject::KartObject(KartParam *param) {
+    m_pointers.m_param = param;
+}
 
 KartObject::~KartObject() {
-    delete m_param;
+    delete m_pointers.m_param;
+}
+
+KartBody *KartObject::createBody(KartPhysics *physics) {
+    return new KartBodyKart(physics);
+}
+
+void KartObject::init() {
+    createSub();
+    auto *physics = KartPhysics::Create(*m_pointers.m_param);
+    auto *body = createBody(physics);
+    m_pointers.m_body = body;
+}
+
+void KartObject::initImpl() {
+    m_pointers.m_sub->init();
 }
 
-void KartObject::createComponents() {}
+void KartObject::prepare() {
+    EGG::Vector3f euler_angles_deg, position;
+
+    System::RaceManager::Instance()->findKartStartPoint(position, euler_angles_deg);
+    m_pointers.m_move->setInitialPhysicsValues(position, euler_angles_deg);
```

Comment:
TODO: verify this works (this should be valid in phase 3)
```suggestion
    move()->setInitialPhysicsValues(position, euler_angles_deg);
```